### PR TITLE
Add capabilities in core api group

### PR DESCRIPTION
**Pull request by yharish991**
### What this PR does / why we need it
This PR
1. Deprecates Capabilities API in run API group
2. Adds Capabilities API in core API group
3. Add a controller to reconcile on Capabilities API in core API group
4. Update capabilities package

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
1. Ran the controller in local machine to verify functionality of capabilities API in core API group
2. Installed the new capabilities package in the cluster and tested it.

### Release note

```release-note
1. Deprecates Capabilities API in run API group
2. Adds new Capabilities API in core API group
```

### CODE REVIEW

File: apis/core/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include ../../common.mk
+# set root for manifests
+RELATIVE_ROOT=.
+
+.PHONY: manifests
+manifests: ## Generate manifests e.g. CRD, RBAC etc.
+	$(MAKE) generate-manifests CONTROLLER_GEN_SRC=./...
```
Comment (codegold79) [inline apis/core/Makefile]: looks like new line is needed here

File: apis/core/v1alpha1/capabilities_types.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
```
Comment (codegold79) [inline apis/core/v1alpha1/capabilities_types.go:1]: nit: copyright year check

File: capabilities/controller/pkg/capabilities/core/capability_controller.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	log := r.Log.WithValues("capability", req.NamespacedName)
+	log.Info("Starting reconcile")
+
+	capability := &corev1alpha1.Capability{}
+	if err := r.Get(ctxCancel, req.NamespacedName, capability); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	var serviceAccountName, namespaceName string
+	// use the default service account when the serviceAccountName is not provided as part of the spec
```
Comment (codegold79) [inline capabilities/controller/pkg/capabilities/core/capability_controller.go:46]:
```suggestion
	// Use the default service account when the service account name is not provided as part of the spec.
```

File: capabilities/controller/pkg/capabilities/core/capability_controller.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	if len(capability.Spec.ServiceAccountName) > 0 {
+		serviceAccountName = capability.Spec.ServiceAccountName
+		namespaceName = req.Namespace
+	} else {
+		serviceAccountName = constants.ServiceAccountWithDefaultPermissions
+		namespaceName = constants.CapabilitiesControllerNamespace
+	}
+	config, err := config.GetConfigForServiceAccount(ctx, r.Client, namespaceName, serviceAccountName, r.Host)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("unable to get config for ClusterQueryClient creation: %w", err)
```
Comment (codegold79) [inline capabilities/controller/pkg/capabilities/core/capability_controller.go:56]: nit: I think error message would be enough to be like,
```fmt.Errorf("get config for ClusterQueryClient creation: %w", err)```

My thinking is, as it tries to get a config, an error will appear why it couldn't be got. Saying "unable to" seems redundant. The following error message indicates the config was unable to be gotten.

Same goes for line 60. Perhaps, "unable to create..." could be simple, "create ClusterQueryClient:".
But this is a matter of preference. It's optional.

File: config/rbac/capabilities_editor_role.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# permissions for end users to edit capabilities.
```
Comment (codegold79) [inline config/rbac/capabilities_editor_role.yaml:1]:
```suggestion
# Permissions for end users to edit capabilities.
```

File: config/rbac/capabilities_viewer_role.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# permissions for end users to view capabilities.
```
Comment (codegold79) [inline config/rbac/capabilities_viewer_role.yaml:1]:
```suggestion
# Permissions for end users to view capabilities.
```

**Review by codegold79 — CHANGES_REQUESTED** [review]
I couldn't find any major stuff--looks good! Only minor comments.

File: capabilities/controller/pkg/capabilities/core/capability_controller.go (hunk 3 above)
Comment (yharish991) [reply capabilities/controller/pkg/capabilities/core/capability_controller.go:46]: `serviceAccountName` is the field name in the spec

File: config/rbac/capabilities_editor_role.yaml (hunk 5 above)
Comment (yharish991) [reply config/rbac/capabilities_editor_role.yaml:1]: these are generated config

File: config/rbac/capabilities_viewer_role.yaml (hunk 6 above)
Comment (yharish991) [reply config/rbac/capabilities_viewer_role.yaml:1]: same, this is generated yaml

File: capabilities/controller/pkg/capabilities/core/capability_controller.go (hunk 4 above)
Comment (rajathagasthya) [reply capabilities/controller/pkg/capabilities/core/capability_controller.go:56]: I prefer using `failed to` or `unable to` phrases in error messages as it is slightly more clear and doesn't feel like an incomplete sentence.

**Review by codegold79 — APPROVED** [review]

File: apis/run/v1alpha1/capability_types.go [hunk 7]
```diff
@@ -149,6 +149,7 @@ type Result struct {
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
+//+kubebuilder:deprecatedversion:warning="Capabilities API in run API group is deprecated, use Capabilities API in core API group"
```
Comment (rajathagasthya) [inline apis/run/v1alpha1/capability_types.go:152]: Let's use full names for API groups.

```
Capability API in run.tanzu.vmware.com is deprecated. Use Capability API from core.tanzu.vmware.com instead.
```

**Review by rajathagasthya — APPROVED** [review]
Minor comment about deprecation message. Will need to regenerate CRD manifest though.

File: apis/run/v1alpha1/capability_types.go (hunk 7 above)
Comment (yharish991) [reply apis/run/v1alpha1/capability_types.go:152]: makes sense, will update

**Pull request merged**